### DISCUSSION TITLE
Set Taxonomy default value in SPO column

### DISCUSSION THREAD

**Discussion by davidgr27** [pnp/powershell#2260] [General]
Hello 
      I have an script for create some columns in a document library in Sharepoint Online and I assign a default value for these columns:
                     Add-PnPTaxonomyField -DisplayName "FunctionalArea" -InternalName "FunctionalArea" -TaxonomyItemId "bc249ecd-69b4-4900-a46b-74e94240c089" -List  "Public" -Required -AddToDefaultView
  
                    $Functional = "2;#Functional Area|0c5e9a11-3414-46dd-898e-c9d4ff02db46"
  
                    Set-PnPField -List $PublicLibrary -Identity "FunctionalArea" -Values @{DefaultValue = $Functional}

          When I've executed my script I don't receive any error message and I check the column configuration and it's correct with the default value but when I try to upload any document to this library the column doesn't have any value, can you give me any clue¿?

          Many thanks.

**Comment by BaronSparky** [comment]
Is the term index number correct on all site collections as held in the hidden taxonomy list?

I would substitute it for "-1" ($Functional = "-1;#Functional Area|0c5e9a11-3414-46dd-898e-c9d4ff02db46")

Or try https://pnp.github.io/powershell/cmdlets/Set-PnPDefaultColumnValues.html

**Comment by davidgr27** [comment]
Hello @BaronSparky 
       Many thanks for your reply, yesterday I tried your suggestion with the ID but when I try to upload any document in the library, the columns not apply the default value until not edit the item properties.
       My last test was to change by SET-PNPDEFAULTCOLUMNSVALUES but when I execute this command, I don't have any error message and I upload the document and it's correct but when I access to the column configuration, the DEFAULT VALUE is blank, it's normal?

**Reply by BaronSparky** [reply]
I have seen this before with list columns to be honest. The MMS field has 2 elements, the display bit you can see with the label and a "Note" field behind it.

I typically add site columns to the site, set the default, and then add that column to a list/view. This is reliable.

Alternatively, I use a PnP template (site or tenant template) and just use the "Default" property with the "-1;#<label>|<guid>" which always works. i.e.,:
`                <Field ID="{guid-guid-guid-guid-guid}" Name="SCategory" DisplayName="S Category" Description="S Categories to describe,..." Group="S Columns" Type="TaxonomyFieldTypeMulti" Mult="FALSE" Required="TRUE">
                    <Default>-1;#<label>|<guid></Default>
                   <Customization>....`

**Reply by davidgr27** [reply]
Hello @BaronSparky 
I understand but we have many Teams sites and every site has different default value in the column, I've surprised because  I've used my script and I don't have any problem but 2 weeks ago I've started to have this problem.

**Reply by BaronSparky** [reply]
You can parameterise the template, so each time you call the template, you can provide a unique value using "{parameter:...}"

Can you have a look at what the list `...FieldValue.FunctionalArea` looks like. It should hopefully have the "Label" and "TermGuid" fields populated.
![image](https://user-images.githubusercontent.com/11838512/185935253-86573837-cfd5-441a-866d-b3effb5a53c1.png)

Your code works for me on newly uploaded documents:
`Add-PnPTaxonomyField -Connection $provSiteConnection -DisplayName "Functional Area" -InternalName "FunctionalArea4" -TaxonomyItemId "d3a098ea-5e27-4710-9ece-606d16fc6c48" -List "Shared Documents" -Required -AddToDefaultView
`
`$Functional = "-1;#Security Assessment|5dee7e20-131f-4b2e-8bfc-0c52bca8e577"
`
`Set-PnPField -Connection $provSiteConnection -List "Shared Documents" -Identity "FunctionalArea4" -Values @{DefaultValue = $Functional}`
